### PR TITLE
Added utility functions to the new NavigationServer:

### PR DESCRIPTION
**Pull request by AndreaCatania**
Re-added functions, to the `NavigationServer`:
```
- Vector3 get_closest_point_to_segment(const Vector3 &p_from, const Vector3 &p_to, const bool &p_use_collision = false);
- Vector3 get_closest_point(const Vector3 &p_point);
- Vector3 get_closest_point_normal(const Vector3 &p_point);
- Object *get_closest_point_owner(const Vector3 &p_point);
```

Hey @mourning20s check this.

### CODE REVIEW

**Comment by mourning20s** [comment]
@AndreaCatania I tried it out and it seems to work as expected. Once again, thank you for taking the time to do this.

**Comment by akien-mga** [comment]
Could you add the relevant documentation too?

**Comment by AndreaCatania** [comment]
> Could you add the relevant documentation too?

Done

**Comment by AndreaCatania** [comment]
@akien-mga any news about this?

File: doc/classes/Navigation.xml [hunk 1]
```diff
@@ -28,6 +28,46 @@
...
+				Returns the nearest point on the navigation mesh surface nearest to the submitted point.
+			</description>
+		</method>
+		<method name="get_closest_point_normal" qualifiers="const">
+			<return type="Vector3">
+			</return>
+			<argument index="0" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the normal for the point returned by [code]get_closest_point[/code].
```
Comment (akien-mga) [inline doc/classes/Navigation.xml]: Use `[method get_closest_point]` instead to generate a hyperlink. Same below.

File: doc/classes/Navigation.xml [hunk 2]
```diff
@@ -28,6 +28,46 @@
...
+				Returns the closest point between the navigation surface and the segment.
+			</description>
+		</method>
+		<method name="get_closest_point" qualifiers="const">
+			<return type="Vector3">
+			</return>
+			<argument index="0" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the nearest point on the navigation mesh surface nearest to the submitted point.
```
Comment (akien-mga) [inline doc/classes/Navigation.xml]: "nearest point ... nearest to the submitted point" sounds unclear.

Comment (akien-mga) [reply doc/classes/Navigation.xml]: Maybe "Returns the point closest to the provided [code]point[/code] on the navigation mesh surface."

File: doc/classes/Navigation2DServer.xml [hunk 3]
```diff
@@ -188,6 +188,28 @@
...
+		</method>
+		<method name="map_get_closest_point_owner" qualifiers="const">
+			<return type="RID">
+			</return>
+			<argument index="0" name="map" type="RID">
+			</argument>
+			<argument index="1" name="point" type="Vector2">
+			</argument>
+			<description>
+				Returns the owner region RID for the point returned by [code]get_closest_point[/code].
```
Comment (akien-mga) [inline doc/classes/Navigation2DServer.xml]: Use `[method map_get_closest_point]`.

File: doc/classes/Navigation2DServer.xml [hunk 4]
```diff
@@ -188,6 +188,28 @@
...
 		</method>
+		<method name="map_get_closest_point" qualifiers="const">
+			<return type="Vector2">
+			</return>
+			<argument index="0" name="map" type="RID">
+			</argument>
+			<argument index="1" name="point" type="Vector2">
+			</argument>
+			<description>
+				Returns the nearest point on the navigation mesh surface nearest to the submitted point.
```
Comment (akien-mga) [inline doc/classes/Navigation2DServer.xml]: Same comment as for the `Navigation2D` equivalent.

File: doc/classes/NavigationServer.xml [hunk 5]
```diff
@@ -188,6 +188,54 @@
...
+		</method>
+		<method name="map_get_closest_point" qualifiers="const">
+			<return type="Vector3">
+			</return>
+			<argument index="0" name="map" type="RID">
+			</argument>
+			<argument index="1" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the nearest point on the navigation mesh surface nearest to the submitted point.
```
Comment (akien-mga) [inline doc/classes/NavigationServer.xml]: Same as above.

File: doc/classes/NavigationServer.xml [hunk 6]
```diff
@@ -188,6 +188,54 @@
...
+		</method>
+		<method name="map_get_closest_point_normal" qualifiers="const">
+			<return type="Vector3">
+			</return>
+			<argument index="0" name="map" type="RID">
+			</argument>
+			<argument index="1" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the normal for the point returned by [code]get_closest_point[/code].
```
Comment (akien-mga) [inline doc/classes/NavigationServer.xml]: `[method map_get_closest_point]`.

File: doc/classes/NavigationServer.xml [hunk 7]
```diff
@@ -188,6 +188,54 @@
...
+		</method>
+		<method name="map_get_closest_point_owner" qualifiers="const">
+			<return type="RID">
+			</return>
+			<argument index="0" name="map" type="RID">
+			</argument>
+			<argument index="1" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the owner region RID for the point returned by [code]get_closest_point[/code].
```
Comment (akien-mga) [inline doc/classes/NavigationServer.xml]: `[method map_get_closest_point]`.

Comment (AndreaCatania) [reply doc/classes/NavigationServer.xml]: done

File: doc/classes/NavigationServer.xml (hunk 6 above)
Comment (AndreaCatania) [reply doc/classes/NavigationServer.xml]: done

File: doc/classes/NavigationServer.xml (hunk 5 above)
Comment (AndreaCatania) [reply doc/classes/NavigationServer.xml]: done

File: doc/classes/Navigation2DServer.xml (hunk 4 above)
Comment (AndreaCatania) [reply doc/classes/Navigation2DServer.xml]: done

File: doc/classes/Navigation2DServer.xml (hunk 3 above)
Comment (AndreaCatania) [reply doc/classes/Navigation2DServer.xml]: done

File: doc/classes/Navigation.xml (hunk 2 above)
Comment (AndreaCatania) [reply doc/classes/Navigation.xml]: done

File: doc/classes/Navigation.xml (hunk 1 above)
Comment (AndreaCatania) [reply doc/classes/Navigation.xml]: done

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**